JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java

```diff
@@ -237,39 +230,39 @@ void deleteDialog(MouseEvent mouseEvent) {
     }
 
     public void generateRawDataTable() {
-        TableView<ObservableList<String>> table = new TableView();
+        TableView<ObservableList<String>> table = new TableView<>();
         table.getItems().clear();
         Timeline timeline = new Timeline(new KeyFrame(Duration.millis(3000), e -> {
             ObservableList<AssetAttribute> attributes = FXCollections.observableArrayList(assetDAOImpl.createAssetInfo(system.getId()).getAssetAttributes());
 
             int columnIndex = 1;
-            TableColumn[] tableColumns = new TableColumn[attributes.size() + 1];
+            TableColumn<ObservableList<String>, String>[] tableColumns = new TableColumn[attributes.size() + 1];
 
-            ArrayList<ArrayList<Measurement>> data = new ArrayList();
+            ArrayList<List<Measurement>> data = new ArrayList<>();
```

Comment:
why change the arraylist to a list?

Reply:
The reason is the following: implementing classes must be used to instantiate new collections, but the result of an instantiation should ideally be stored in a variable whose type is a Java Collection interface

Reply:
ok